crates/trigger/src/cli/initial_kv_setter.rs

```diff
@@ -23,6 +23,9 @@ impl<F: RuntimeFactors, U> ExecutorHooks<F, U> for InitialKvSetterHook {
         &mut self,
         configured_app: &spin_factors::ConfiguredApp<F>,
     ) -> anyhow::Result<()> {
+        if self.kv_pairs.is_empty() {
+            return Ok(());
+        }
         let Some(kv) = configured_app.app_state::<KeyValueFactor>().ok() else {
             return Ok(());
         };
```

Comment:
It seems like this should actually be an error now... :shrug:

Reply:
Maybe some context e.g. "you tried to set some kv but no kv!"

Reply:
Done